### ISSUE TITLE
Patch Request: "Don't Play Music Videos" option.

### ISSUE THREAD

**Issue by NarrikSynthfox** [ReVanced/revanced-patches#1619]
### Application

Youtube Music

### Issue

For premium users, theres an option that makes it so that it doesn't play the music video version of songs when only playing audio. If it's possible to allow that to be used tor non-premium users, that would be great. Some songs, the music videos have long intros or sfx over the music that isn't in the streaming upload of the song.

### Patch

Show the option for "Don't play music videos" (I think thats what it's labeled as) for all users.

### Motivation

I have seen a lot of people requesting SponsorBlock for Youtube Music, and most of the reasons why would be solved with this patch. Hopefully its possible to accomplish.

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [x] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by oSumAtrIX** [comment]
The stream the API sends you can not be changed.

**Comment by NarrikSynthfox** [comment]
Yeah, but there is an option that is built in natively to YouTube Music, that only shows up for premium users, and I was wondering if it were possible to make it show up for non-premium as well somehow.

**Comment by oSumAtrIX** [comment]
I don't think so. The API will only work if the user is known to have premium.

**Comment by ILoveOpenSourceAppications** [comment]
Well I don't know about the premium part, but I'm pretty sure that RVX has a patch which does this.

**Comment by oSumAtrIX** [comment]
Please link the relevant commits.

**Comment by ILoveOpenSourceAppications** [comment]
![Screenshot_20230212-183723~2](https://user-images.githubusercontent.com/117499019/218312882-4fa8f6a4-648a-42ee-9d32-6ade892c6f1f.png)
This is the result of the patch. I can't seem to find the commit though.

**Comment by oSumAtrIX** [comment]
You can use "git blame" to find the commit to the code which enables it, please then send a commit link.

**Comment by ILoveOpenSourceAppications** [comment]
How do I do this exactly? Not very used to GitHub actually.

**Comment by oSumAtrIX** [comment]
[This](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file) and [this](https://git-scm.com/docs/git-blame) might be helpful for you.

**Comment by ILoveOpenSourceAppications** [comment]
I have a doubt. Don't we have exclusive audio playback option? Doesn't it play just the audio only? I think the one I mentioned above does the same only.

**Comment by oSumAtrIX** [comment]
I believe so too

**Comment by NarrikSynthfox** [comment]
It does play audio only, but it's the audio for the music video.

**Comment by ILoveOpenSourceAppications** [comment]
Sorry for the high hopes if I gave you any @NarrikSynthfox, but I don't think there's any patch atm which purely shifts from video mode to audio mode of the same song.

**Comment by HMartusewiczVox** [comment]
I recently discovered this projects and they've done it:
https://github.com/vfsfitvnm/ViMusic/issues/88
